apps/backend/src/user/user.service.ts

```diff
@@ -101,4 +102,38 @@ export class UserService {
       throw new NotFoundException(`User with id ${id} not found`);
     }
   }
+
+  async saveProfilePicture(authSchId: string, buffer: Buffer, mimetype: string) {
+    if (mimetype !== 'image/png' && mimetype !== 'image/jpeg') {
+      throw new BadRequestException('Invalid image format');
+    }
+    try {
+      await this.createOrUpdateProfilePicture(authSchId, buffer, mimetype);
+    } catch (error) {
+      throw new NotFoundException(`User with id ${authSchId} not found`);
+    }
+  }
+
+  async findProfilePicture(authSchId: string): Promise<Buffer> {
+    try {
+      const profilePic = await this.prisma.profilePicture.findUnique({ where: { userId: authSchId } });
+      return profilePic.profileImage;
+    } catch (error) {
+      throw new NotFoundException(`User with id ${authSchId} not found`);
```

Comment:
I guess this might work, but it!s not too elegant. findUnique returns null when it finds nothing (null.profileImage probably throws an error, but you should use findUniqueOrThrow)

Reply:
Weird, I did it this way, because I didn't get a typescript error